CorsixTH/Lua/dialogs/edit_room.lua

```diff
@@ -1209,6 +1209,25 @@ local function doorWallOffsetCalculations(x, y, wall)
   return x, y, x_mod, y_mod, wall
 end
 
+-- this is a pseudo bitwise OR operation
+-- assumes value2 is always a power of 2 (limits carry errors in the addition)
+-- mimics the logic of hasBit with the addition if bit not set
+--!param value1 (int) value to check set bit of
+--!param value2 (int) power of 2 value - bit enumeration
+--!return (int) value1 and value2 'bitwise' or.
+local function bitOr(value1, value2)
+  return value1 % (value2 + value2) >= value2 and value1 or value1 + value2
+end
+
+--! Check bit is set
+--!param value (int) value to check set bit of
+--!param bit (int) 0-base index of bit to check
+--!return (boolean) true if bit is set.
+local function hasBit(value, bit)
+  local p = 2 ^ bit
+  return value % (p + p) >= p
+end
+
```

Comment:
Obviously, lua doesn't have proper bit operators so arguably squeezing 3 bits into an integer isn't very lua-like.

Another solution would be to convert the bit-stuff to an array of 3 booleans (invalid_tile uses bits 1, 2, and 3 apparently).

A second alternative could be to implement the bit functions in c++, but that would still keep the bit-stuff of invalid_tile.